gin/jose.go

```diff
@@ -65,6 +66,44 @@ func TokenSigner(hf ginkrakend.HandlerFactory, logger logging.Logger) ginkrakend
 	}
 }
 
+func GetRefreshedToken(cfg *krakendjose.SignatureConfig, logger logging.Logger, r *http.Request) (*jwt.JSONWebToken, string, error) {
+	cookie, err := r.Cookie(cfg.RefreshCookieKey)
+
+	if err != nil {
+		logger.Warning("JOSE: refresh token not set when attempting to refresh")
+		return nil, "", jwt.ErrExpired
+	}
+
+	_, err = jwt.ParseSigned(cookie.Value)
+
+	if err != nil {
+		logger.Warning("JOSE: refresh token signature is invalid")
+		return nil, "", err
+	}
+
+	client := &http.Client{}
```

Comment:
I think here we can reuse `http.Client` (and it can be used concurrently)

Reply:
Thanks ! Didn't know that. I tried it and it worked